typescript/packages/plugins/mode-governance/src/mode-governance.service.ts

```diff
@@ -0,0 +1,110 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { parseUnits, formatUnits } from "viem";
+import { VOTING_ESCROW_ABI } from "./abi";
+import {
+    MODE_VOTING_ESCROW,
+    BPT_VOTING_ESCROW,
+} from "./constants";
+import {
+    StakeParameters,
+    GetStakeInfoParameters,
+    GetBalanceParameters,
+} from "./parameters";
+
+export class ModeGovernanceService {
+    @Tool({
+        description: "Stake MODE or BPT tokens in the Mode governance system. Requires MODE or BPT tokens to be approved first.",
+    })
+    async stakeTokens(walletClient: EVMWalletClient, parameters: StakeParameters) {
```

Comment:
Name is too generic.

Maybe `stake_tokens_for_mode_governance`

Reply:
correct